json11.cpp

```diff
@@ -364,13 +365,61 @@ struct JsonParser {
             i++;
     }
 
+    /* consume_comment()
+     *
+     * Advance comments (c-style inline and multiline).
+     */
+    bool consume_comment() {
+      bool comment_found = false;
+      if (str[i] == '/') {
+        i++;
+        if (str[i] == '/') { // inline comment
+          i++;
+          // advance until next line
+          while (str[i] != '\n')
+            i++;
+          comment_found = true;
+        }
+        else if (str[i] == '*') { // multiline comment
+          i++;
+           // advance until closing tokens
+          while (!(str[i] == '*' && str[i+1] == '/')) {
+            if (i == str.size())
```

Comment:
By the time check runs you've already accessed str[i] at least (which probably works in your current tests due to null termination).  Need to check size earlier.